pkg/kademlia/kademlia_test.go

```
@@ -5,251 +5,164 @@ package kademlia
-	p++
-
-	err = boot.Bootstrap(context.Background())
-	assert.NoError(t, err)
-	for i := 0; i < testNetSize; i++ {
-		gg := strconv.Itoa(p)
-
-		nid, err := newID()
-		assert.NoError(t, err)
-		id := NodeID(nid)
-
-		dht, err := NewKademlia(&id, []overlay.Node{bootNode}, ip, gg)
-		assert.NoError(t, err)
+func TestNewKademlia(t *testing.T) {
+	cases := []struct {
+		id          dht.NodeID
+		bn          []proto.Node
+		addr        string
+		expectedErr error
+	}{
+		{
+			id: func() *node.ID {
+				id, err := node.NewID()
+				assert.NoError(t, err)
+				return id
+			}(),
+			bn:   []proto.Node{proto.Node{Id: "foo"}},
+			addr: "127.0.0.1:8080",
+		},
+	}
 
-		p++
-		dhts = append(dhts, dht)
-		err = dht.ListenAndServe()
+	for _, v := range cases {
+		ca, err := provider.NewCA(ctx, 12, 4)
 		assert.NoError(t, err)
-		err = dht.Bootstrap(context.Background())
+		identity, err := ca.NewIdentity()
 		assert.NoError(t, err)
-
+		actual, err := NewKademlia(v.id, v.bn, v.addr, identity)
+		assert.Equal(t, v.expectedErr, err)
+		assert.Equal(t, actual.bootstrapNodes, v.bn)
+		assert.Equal(t, actual.stun, true)
+		assert.NotNil(t, actual.nodeClient)
+		assert.NotNil(t, actual.routingTable)
 	}
-
-	return dhts, bootNode
 }
 
-func newTestKademlia(t *testing.T, ip, port string, b overlay.Node) *Kademlia {
-	i, err := newID()
+func TestLookup(t *testing.T) {
+	lis, err := net.Listen("tcp", ":0")
 	assert.NoError(t, err)
-	id := NodeID(i)
-	n := []overlay.Node{b}
-
-	kad, err := NewKademlia(&id, n, ip, port)
-	assert.NoError(t, err)
-	return kad
-}
-
-func TestBootstrap(t *testing.T) {
-	t.Skip()
-	dhts, bootNode := bootstrapTestNetwork(t, "127.0.0.1", "3000")
-
-	defer func(d []dht.DHT) {
-		for _, v := range d {
-			_ = v.Disconnect()
-		}
-	}(dhts)
 
-	cases := []struct {
-		k *Kademlia
-	}{
-		{
-			k: newTestKademlia(t, "127.0.0.1", "2999", bootNode),
-		},
-	}
+	srv, mns := newTestServer([]*proto.Node{&proto.Node{Id: "foo"}})
+	go func() { _ = srv.Serve(lis) }()
```

Don't ignore the error here.